From 8b7a012c480bc75f6da314d5e3279f954b750d20 Mon Sep 17 00:00:00 2001
From: Meron Cohen <mecohen@habana.ai>
Date: Thu, 08 May 2025 18:43:54 +0300
Subject: [PATCH] [SW-227605] g2e: fit: enable bis-e-compliant

Create a FIT for enabling BIS-E-Compliant variant.
(eFuse address 0x3A, bit #27).

Change-Id: Iffd58659245d947d5156d6b8723e1a64e8324797
Signed-off-by: Meron Cohen <mecohen@habana.ai>
---

diff --git a/scripts/make/gaudi2/Makefile b/scripts/make/gaudi2/Makefile
index ced798f..a7e2e49 100644
--- a/scripts/make/gaudi2/Makefile
+++ b/scripts/make/gaudi2/Makefile
@@ -230,6 +230,7 @@
 	${HL_DEV}-eeprom-recovery.itb	\
 	${HL_DEV}-bis_compliant_enable.itb	\
 	${HL_DEV}-bis_d_compliant_enable.itb	\
+	${HL_DEV}-bis_e_compliant_enable.itb	\
 	${HL_DEV}-erom_recovery_enable.itb	\
 	${HL_DEV}-erom_erase.itb		\
 	${HL_DEV}-secureboot_config.itb		\
@@ -243,6 +244,7 @@
 				gaudi2-eeprom-create-recovery	\
 				gaudi2-bis-compliant-enable-create \
 				gaudi2-bis-d-compliant-enable-create \
+				gaudi2-bis-e-compliant-enable-create \
 				gaudi2-erom-recovery-enable-create	\
 				gaudi2-erom-erase-create gaudi2-secureboot-config-create \
 				gaudi2-secureboot-config-dbg-create \
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index 19f9bf8..d5b620b 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -191,6 +191,16 @@
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
 		-sshpass -p$(HL_PASSWD) $(SCP) $(ITBS_DIR)/${DEV_PREFIX}-bis_d_compliant_enable.itb root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(DEV_PREFIX)/)
 
+.PHONY: %-bis-e-compliant-enable-create
+%-bis-e-compliant-enable-create:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh bis_e_compliant_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader no-load-only)
+
+.PHONY: %-bis-e-compliant-enable-install
+%-bis-e-compliant-enable-install:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		-sshpass -p$(HL_PASSWD) $(SCP) $(ITBS_DIR)/${DEV_PREFIX}-bis_e_compliant_enable.itb root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(DEV_PREFIX)/)
+
 .PHONY: %-erom-recovery-enable-create
 %-erom-recovery-enable-create:
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
